ttnn/cpp/ttnn/operations/moreh/moreh_group_norm/device/moreh_group_norm_device_operation.cpp

```diff
@@ -0,0 +1,170 @@
+
+    Shape mean_rstd_shape = Shape(tt::tt_metal::LegacyShape(mean_rstd_origin_shape, mean_rstd_padding));
+    return {output_shape, mean_rstd_shape, mean_rstd_shape};
+}
+
+MorehGroupNormOperation::tensor_return_value_t MorehGroupNormOperation::create_output_tensors(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    const auto output_shapes = compute_output_shapes(operation_attributes, tensor_args);
+    auto dtype = tensor_args.input.get_dtype();
+    Layout layout{Layout::TILE};
+    auto device = tensor_args.input.device();
+
+    std::vector<std::optional<Tensor>> result;
+    result.reserve(3);
+
+    // output
+    if (tensor_args.output_tensors[0].has_value()) {
+        result.push_back(tensor_args.output_tensors[0].value());
+    } else {
+        result.push_back(
+            create_device_tensor(output_shapes[0].value(), dtype, layout, device, operation_attributes.memory_config));
+    }
+
+    // mean
+    if (tensor_args.output_tensors[1].has_value()) {
+        result.push_back(tensor_args.output_tensors[1].value());
+    } else if (operation_attributes.are_required_outputs[1]) {
+        result.push_back(create_device_tensor(
+            output_shapes[1].value(), dtype, layout, device, operation_attributes.mean_memory_config));
+    } else {
+        result.push_back(std::nullopt);
+    }
+
+    // rstd
+    if (tensor_args.output_tensors[2].has_value()) {
+        result.push_back(tensor_args.output_tensors[2].value());
+    } else if (operation_attributes.are_required_outputs[2]) {
+        result.push_back(create_device_tensor(
+            output_shapes[2].value(), dtype, layout, device, operation_attributes.rstd_memory_config));
+    } else {
+        result.push_back(std::nullopt);
+    }
+    return std::move(result);
+}
+
+std::tuple<MorehGroupNormOperation::operation_attributes_t, MorehGroupNormOperation::tensor_args_t>
+MorehGroupNormOperation::invoke(
+    const Tensor& input,
+    const uint32_t num_groups,
+    const float eps,
+    const std::optional<const Tensor> gamma,
+    const std::optional<const Tensor> beta,
+    const std::vector<bool>& are_required_outputs,
+    const std::optional<const Tensor> output,
+    const std::optional<const Tensor> mean,
+    const std::optional<const Tensor> rstd,
+    const std::optional<MemoryConfig>& memory_config,
+    const std::optional<MemoryConfig>& mean_memory_config,
+    const std::optional<MemoryConfig>& rstd_memory_config,
+    const std::optional<DeviceComputeKernelConfig>& compute_kernel_config) {
+    operation_attributes_t operation_attributes{
+        num_groups,
+        eps,
+        are_required_outputs,
+        memory_config.value_or(operation::DEFAULT_OUTPUT_MEMORY_CONFIG),
+        mean_memory_config.value_or(operation::DEFAULT_OUTPUT_MEMORY_CONFIG),
+        rstd_memory_config.value_or(operation::DEFAULT_OUTPUT_MEMORY_CONFIG),
```

Comment:
```suggestion
        memory_config.value_or(input.memory_config()),
        mean_memory_config.value_or(input.memory_config()),
        rstd_memory_config.value_or(input.memory_config()),
```
DEFAULT_OUTPUT_MEMORY_CONFIG mean DRAM and interleved layout.
Using the default memory config can cause incorrect behavior where DRAM output is produced when L1 input is provided. In such cases, the memory config of the input should be used."